src/java/org/apache/cassandra/index/IndexRegistry.java

```diff
@@ -189,7 +203,50 @@ public Searcher searcherFor(ReadCommand command)
             }
         };
 
-        public void registerIndex(Index index)
+        Index.Group group = new Index.Group() {
+            @Override
+            public Set<Index> getIndexes()
+            {
+                return Collections.singleton(index);
+            }
+
+            @Override
+            public void addIndex(Index index)
+            {
+            }
+
+            @Override
+            public void removeIndex(Index index)
+            {
+            }
+
+            @Override
+            public boolean containsIndex(Index i)
+            {
+                return index == i;
+            }
+
+            @Override
+            public Index.Indexer indexerFor(Predicate<Index> indexSelector, DecoratedKey key, RegularAndStaticColumns columns, int nowInSec, WriteContext ctx, IndexTransaction.Type transactionType, Memtable memtable)
```

Comment:
Should this and `getFlushObserver` be `Nullable` as well?

Reply:
+1